src/services/refactors/extractSymbol.ts

```diff
@@ -23,25 +23,38 @@ namespace ts.refactor.extractSymbol {
 
         const functionActions: RefactorActionInfo[] = [];
         const usedFunctionNames: Map<boolean> = createMap();
+        let innermostErrorFunctionAction: ts.RefactorActionInfo | undefined;
 
         const constantActions: RefactorActionInfo[] = [];
         const usedConstantNames: Map<boolean> = createMap();
+        let innermostErrorConstantAction: ts.RefactorActionInfo | undefined;
+        
 
         let i = 0;
         for (const {functionExtraction, constantExtraction} of extractions) {
-            // Skip these since we don't have a way to report errors yet
+            const description = functionExtraction.description;
             if (functionExtraction.errors.length === 0) {
                 // Don't issue refactorings with duplicated names.
                 // Scopes come back in "innermost first" order, so extractions will
                 // preferentially go into nearer scopes
-                const description = functionExtraction.description;
                 if (!usedFunctionNames.has(description)) {
                     usedFunctionNames.set(description, true);
                     functionActions.push({
                         description,
                         name: `function_scope_${i}`
                     });
                 }
+            } else if (!innermostErrorFunctionAction) {

```

Comment:
I guess the "innermost" error is the only one we care about?

Reply:
Yep, this is probably the most relevant since it's lexically closest to problem.